fdbserver/DataDistribution.actor.cpp

```diff
@@ -4533,13 +4600,47 @@ ACTOR Future<Void> keyValueStoreTypeTracker(DDTeamCollection* self, TCServerInfo
 	    wait(brokenPromiseToNever(server->lastKnownInterface.getKeyValueStoreType.getReplyWithTaskID<KeyValueStoreType>(
 	        TaskPriority::DataDistribution)));
 	server->storeType = type;
-
 	if (type != self->configuration.storageServerStoreType) {
 		if (self->wrongStoreTypeRemover.isReady()) {
 			self->wrongStoreTypeRemover = removeWrongStoreType(self);
 			self->addActor.send(self->wrongStoreTypeRemover);
 		}
 	}
+	return Never();
+}
+
+// Read storage metadata from database; Error is catched by the caller
+ACTOR Future<Void> checkStorageMetadata(DDTeamCollection* self, TCServerInfo* server) {
+	state KeyBackedObjectMap<UID, StorageMetadataType, decltype(IncludeVersion())> metadataMap(serverMetadataKeys.begin,
+	                                                                                           IncludeVersion());
+	state Reference<ReadYourWritesTransaction> tr = makeReference<ReadYourWritesTransaction>(self->cx);
+	state StorageMetadataType data(timer_int());
+	// printf("------ read metadata %s\n", server->id.toString().c_str());
+	// read storage metadata
+	loop {
+		try {
+			tr->setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
+			auto property = metadataMap.getProperty(server->id);
+			Optional<StorageMetadataType> metadata = wait(property.get(tr));
+			// NOTE: in upgrade testing, there may not be any metadata
+			if (metadata.present()) {
+				data = metadata.get();
+			} else {
+				metadataMap.set(tr, server->id, data);
+			}
+			wait(tr->commit());
+			break;
+		} catch (Error& e) {
+			wait(tr->onError(e));
+		}
+	}
+
+	// add server to wiggler
+	if (self->storageWiggler->contains(server->id)) {
```

Comment:
For better understanding; the function does the following:
1. Read database configuration
2. If needed add SS for wiggling OR just update metadata.

Any reason for returning "Never"?

Reply:
Because in `storageTracker` there's an `when()`[ clause](https://github.com/apple/foundationdb/pull/6219/files/2d4d129c59cf527d92fd990e486db994f2d55fa3#diff-f3e66a6ee528a2fb4f9f61766269750c368f639ff9e231b00d6c514bbe5e21a6R5135) wait those two actor (keyValueStoreTracker and checkStorageMetadata). This when() is for catch some thrown exception, actually we don't expect they return if every thing is normal. 

(P.S. This is a classical method to handle exception swallow in Flow.